cypress/integration/trigger_builder_test.js

```diff
@@ -1,11 +1,263 @@
+const triggerConditionToLabel = {
+  device_connected: 'Device Connected',
+  device_disconnected: 'Device Disconnected',
+  device_error: 'Device Error',
+  device_empty_cache_received: 'Empty Cache Received',
+  incoming_data: 'Incoming Data',
+  value_change: 'Value Change',
+  value_change_applied: 'Value Change Applied',
+  path_created: 'Path Created',
+  path_removed: 'Path Removed',
+  value_stored: 'Value Stored',
+};
+
+const triggerOperatorToLabel = {
+  '*': '*',
+  '==': '==',
+  '!=': '!=',
+  '>': '>',
+  '>=': '>=',
+  '<': '<',
+  '<=': '<=',
+  contains: 'Contains',
+  not_contains: 'Not Contains',
+};
+
+const setupTriggerEditorFromSource = (trigger) => {
+  cy.get('#triggerSource')
+    .scrollIntoView()
+    .clear()
+    .type(JSON.stringify(trigger), { parseSpecialCharSequences: false });
+  cy.wait(1500);
+};
+
+const checkTriggerEditorUIValues = (trigger) => {
+  cy.get('#triggerName').scrollIntoView().should('be.visible').and('have.value', trigger.name);
+  cy.get('#triggerCondition')
+    .scrollIntoView()
+    .should('be.visible')
+    .contains(triggerConditionToLabel[trigger.simple_triggers[0].on])
+    .should('be.selected');
+
+  const isDataTrigger = trigger.simple_triggers[0].type === 'data_trigger';
```

Comment:
in the following checks, we only use the first element (index 0) what about assigning it to a named constant?
```
const simpleTrigger = trigger.simple_triggers[0]
```

Reply:
Sure, changed it.